apps/dokploy/components/dashboard/docker/logs/docker-logs-id.tsx

```diff
@@ -1,115 +1,272 @@
+  const [search, setSearch] = React.useState<string>("");
+  const [since, setSince] = React.useState<TimeFilter>("all");
+  const [typeFilter, setTypeFilter] = React.useState<TypeFilter>("all");
+  const scrollRef = useRef<HTMLDivElement>(null);
+
+  const scrollToBottom = () => {
+    if (autoScroll && scrollRef.current) {
+      scrollRef.current.scrollTop = scrollRef.current.scrollHeight;
+    }
+  };
+
+  const handleScroll = () => {
+    if (!scrollRef.current) return;
+
+    const { scrollTop, scrollHeight, clientHeight } = scrollRef.current;
+    const isAtBottom = Math.abs(scrollHeight - scrollTop - clientHeight) < 10;
+    setAutoScroll(isAtBottom);
+  };
+
+  const handleSearch = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setRawLogs("");
+    setFilteredLogs([]);
+    setSearch(e.target.value || "");
+  };
+
+  const handleLines = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setRawLogs("");
+    setFilteredLogs([]);
+    setLines(Number(e.target.value) || 1);
+  };
+
+  const handleSince = (value: TimeFilter) => {
+    setRawLogs("");
+    setFilteredLogs([]);
+    setSince(value);
+  };
+
+  const handleTypeFilter = (value: TypeFilter) => {
+    setTypeFilter(value);
+  };
+
+  useEffect(() => {
+    setRawLogs("");
+    setFilteredLogs([]);
+  }, [containerId]);
+
+  useEffect(() => {
+    if (!containerId) return;
+
+    const protocol = window.location.protocol === "https:" ? "wss:" : "ws:";
+    const params = new globalThis.URLSearchParams({
+      containerId,
+      tail: lines.toString(),
+      since,
+      search,
+    });
+
+    if (serverId) {
+      params.append("serverId", serverId);
+    }
+
+    const wsUrl = `${protocol}//${
+      window.location.host
+    }/docker-container-logs?${params.toString()}`;
+    console.log("Connecting to WebSocket:", wsUrl);
+    const ws = new WebSocket(wsUrl);
+
+    ws.onopen = () => {
+      console.log("WebSocket connected");
+    };
+
+    ws.onmessage = (e) => {
+      // console.log("Received message:", e.data);
+      setRawLogs((prev) => prev + e.data);
+    };
+
+    ws.onerror = (error) => {
+      console.error("WebSocket error:", error);
+    };
+
+    ws.onclose = (e) => {
+      console.log("WebSocket closed:", e.reason);
+    };
+
+    return () => {
+      if (ws.readyState === WebSocket.OPEN) {
+        ws.close();
+      }
+    };
+  }, [containerId, serverId, lines, search, since]);
+
+  const handleDownload = () => {
+    const logContent = filteredLogs
+      .map(
+        ({ timestamp, message }: { timestamp: Date | null; message: string }) =>
+          `${timestamp?.toISOString() || "No timestamp"} ${message}`
+      )
+      .join("\n");
+
+    const blob = new Blob([logContent], { type: "text/plain" });
+    const url = URL.createObjectURL(blob);
+    const a = document.createElement("a");
+    const appName = data.Name.replace("/", "") || "app";
+    a.href = url;
+    a.download = `logs-${appName}-${new Date().toISOString()}.txt`;

```

Comment:
```suggestion
    const isoDate = new Date().toISOString();
    a.download = `${appName}-${serviceName}_${isoDate.slice(0,10).replace(/-/g, '')}_${isoDate.slice(11,19).replace(/:/g, '')}.log.txt`;
```

Changes the format from:

```
logs-testing-outline-625cd2-redis-1-2024-12-11T21_31_33.401Z.txt
```

to:

```
outline-redis_20241211_213133.log.txt
```

Reply:
how do you define the app name, and the serviceName ?

Reply:
Apologies - that was in my code. It should just be your same $appName. I  just wanted to tweak the filename to avoid using characters like `:`.

Reply:
so it will look like :
```
space-x-umami-5eaec5-umami-1_20241211_222314.log.txt
```